src/org/rascalmpl/runtime/$RascalModule.java

```diff
@@ -2885,22 +2885,11 @@ else if (left.length() > right.length()) {
 	}
 	
 	public final IString $str_escape_for_regexp(IString insert) {
-		StringBuilder sw = new StringBuilder();
-		for(int i : insert) {
-			char c = (char) i;
-			switch(c){
-				case '.': sw.append("\\."); break;
-				case '(': sw.append("\\("); break;
-				case ')': sw.append("\\)"); break;
-				case '*': sw.append("\\*"); break;
-				default: sw.append(c);
-			}
-		}
-		return $RVF.string(sw.toString());
+		return $RVF.string(Pattern.quote(insert.getValue()));
```

Comment:
I really do not see how `Pattern.quote` implements the same escaping as done here.

Reply:
It's a feature of the Java regex parser that everything between \Q and \E is taken literally and thus not interpreted as a meta symbol. That goes for every sign, including:

* Groups ( )
* Character classes [ ]
* The any symbol .
* The negation ~
* The or | 
* Lookahead and lookbehind.. >> !>>
* The backslash builtins \w \W etc.
* Etc.